src/Features/LanguageServer/Protocol/Handler/OnAutoInsert/OnAutoInsertHandler.cs

```diff
@@ -48,18 +49,24 @@ internal class OnAutoInsertHandler : AbstractStatelessRequestHandler<LSP.Documen
         public override async Task<LSP.DocumentOnAutoInsertResponseItem?> HandleRequestAsync(LSP.DocumentOnAutoInsertParams autoInsertParams, RequestContext context, CancellationToken cancellationToken)
         {
             var document = context.Document;
-
             if (document == null)
             {
                 return null;
             }
 
             var service = document.GetRequiredLanguageService<IDocumentationCommentSnippetService>();
 
+            // We should use the options passed in by LSP instead of the document's options.
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
are the options passed in by LSP correct?  Like for example if you have an editor config setting for tabs vs spaces.   Does LSP read that?  Similarly for tools options settings.

Reply:
From testing with Razor, it correctly recognizes changes in Tools->Option settings. I'm not sure about editorconfig, will test shortly.

Regardless, I think we need this change since Razor files won't be correctly formatted without it, namely because we need to use Razor's Tools->Options settings instead of C#'s.

Reply:
It does indeed seem to support editorconfig as well (at least in C#, I think I recall editorconfig currently being unsupported for Razor files).

Reply:
> Ah, and I guess with the dynamic file info they provide they don't have a way to set those options on the generated document?

Even if that were the case I wouldn't imagine we'd want to. Given the platform is the owner of settings it should be the one informing you on what settings to use. Out of curiosity does on type formatting or document formatting have this same gap?